rust/oak/src/lib.rs

```diff
@@ -77,12 +83,64 @@ impl ReceiveChannelHalf {
     }
 }
 
-impl Read for ReceiveChannelHalf {
-    fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
-        Ok(unsafe { wasm::channel_read(self.handle, buf.as_mut_ptr(), buf.len()) })
+impl ReceiveChannelHalf {
+    pub fn read_message(&mut self, buf: &mut Vec<u8>) -> std::io::Result<usize> {
+        let result = unsafe { wasm::channel_read(self.handle, buf.as_mut_ptr(), buf.capacity()) };
```

Comment:
Relying on capacity seems a bit brittle, though probably it is fine in practice. How about increase the length as much as possible, and then `truncate` it down to the amount of bytes actually read?

Reply:
I was hoping we would have a number for "as much as possible", and essentially keep around a buffer of that size and reuse it over and over, and clone a prefix of it when handing data out to callers. Anyways I think your approach is just fine, thanks!